src/vivarium_census_prl_synth_pop/results_processing/ssn_and_itin.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any
+
+import numpy as np
+import pandas as pd
+from vivarium.framework.randomness import RandomnessStream
+
+from vivarium_census_prl_synth_pop.utilities import random_integers
+
+
+def generate_itin(
+    size: int,
+    additional_key: Any,
+    randomness: RandomnessStream,
+) -> np.ndarray:
+    """
+    Generate a np.ndarray of length `size` of hopefully unique ITIN values.
+
+        first three digits 900-999, followed by dash
+        next two digits between 50-65, 70-88, 90-92, or 94-99, followed by a dash
+        last four digits between 1-9999
+
+    :param size: The number of itins to generate
+    :param additional_key: Additional key to be used by randomness
+    :param randomness: RandomnessStream stream to use
+
+    """
+    itin = pd.DataFrame(index=pd.Index(range(size)))
+
+    # Area numbers have range 900-999, inclusive
+    itin["area"] = random_integers(
+        min_val=900,
+        max_val=999,
+        index=itin.index,
+        randomness=randomness,
+        additional_key=f"{additional_key}_itin_area",
+    )
+
+    # Group numbers have range 50-65, 70-88, 90-92, or 94-99, inclusive
+    itin["group"] = random_integers(
+        min_val=1,
+        max_val=44,  # (65-50+1)+(88-70+1)+(92-90+1)+(99-94+1)
```

Comment:
I think you can do this much more cleanly by creating a Series mapping to the acceptable values from consecutive integers; that would remove the need for these manual calculations and the ones below.

Reply:
Yeah, agreed, that would be a cleaner and probably more efficient. I'll keep that in mind, I'll be in this code for SSN as well.